integration-tests/src/tests/client/features/delegate_action.rs

```diff
@@ -244,9 +257,119 @@ fn meta_tx_fn_call() {
     })];
 
     let outcome =
-        check_meta_tx_fn_call(node, actions, method_name_len, 0, sender, relayer, receiver);
+        check_meta_tx_fn_call(&node, actions, method_name_len, 0, sender, relayer, receiver);
 
     // Check that the function call was executed as expected
     let fn_call_logs = &outcome.receipts_outcome[1].outcome.logs;
     assert_eq!(fn_call_logs, &vec!["hello".to_owned()]);
 }
+
+#[test]
+fn meta_tx_deploy() {
+    let sender = bob_account();
+    let relayer = alice_account();
+    // Can only deploy on own account
+    let receiver = sender.clone();
+    let node = RuntimeNode::new(&relayer);
+    let fee_helper = fee_helper(&node);
+
+    let code = smallest_rs_contract().to_vec();
+    let tx_cost = fee_helper.deploy_contract_cost(code.len() as u64);
+    let actions = vec![Action::DeployContract(DeployContractAction { code })];
+    check_meta_tx_no_fn_call(&node, actions, tx_cost, 0, sender, relayer, receiver);
+}
+
+#[test]
+fn meta_tx_stake() {
+    let sender = bob_account();
+    let relayer = alice_account();
+    // Can only stake on own account
+    let receiver = sender.clone();
+    let node = RuntimeNode::new(&relayer);
+    let fee_helper = fee_helper(&node);
+
+    let tx_cost = fee_helper.stake_cost();
+    let public_key = create_user_test_signer(&sender).public_key;
+    let actions = vec![Action::Stake(StakeAction { public_key, stake: 0 })];
+    check_meta_tx_no_fn_call(
+        &node,
+        actions,
+        tx_cost,
+        0,
+        sender.clone(),
+        relayer.clone(),
+        receiver.clone(),
+    );
+}
+
+#[test]
+fn meta_tx_add_key() {
+    let sender = bob_account();
+    let relayer = alice_account();
+    // Can only add key on own account
+    let receiver = sender.clone();
+    let node = RuntimeNode::new(&relayer);
+    let fee_helper = fee_helper(&node);
+
+    let tx_cost = fee_helper.add_key_full_cost();
+    // any public key works as long as it doesn't exists on the receiver, the
+    // relayer public key is just handy
+    let public_key = node.signer().public_key();
+    let actions =
+        vec![Action::AddKey(AddKeyAction { public_key, access_key: AccessKey::full_access() })];
+    check_meta_tx_no_fn_call(&node, actions, tx_cost, 0, sender, relayer, receiver);
```

Comment:
can you also check that the key was added. ?
(if we have an easy way of doing so)

Reply:
good call, yes it's pretty easy